tests/jenkins/TestPrintArtifactDownloadUrlsForStaging.groovy

```diff
@@ -6,23 +6,73 @@
  * compatible open source license.
  */
 
-package jenkins.tests
-
+import jenkins.tests.BuildPipelineTest
 import org.junit.*
 
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.core.IsNull.notNullValue
+
+
+class TestPrintArtifactDownloadUrlsForStaging extends BuildPipelineTest implements LibFunctionTester {
+
+    private List artifactFileNames
+    private String uploadPath
 
-class TestPrintArtifactDownloadUrlsForStaging extends BuildPipelineTest {
 
     @Before
     void setUp() {
+
+        this.artifactFileNames = ['dummy_file.tar.gz', 'dummy_file.tar.gz.sig', 'a_dummy_file.tar.gz']
+        this.uploadPath = 'dummy/upload/path'
+
+        this.registerLibTester(new TestPrintArtifactDownloadUrlsForStaging(
+                artifactFileNames: artifactFileNames,
+                uploadPath: uploadPath
+        ))
+
         super.setUp()
-        binding.setVariable('filenamesForUrls', ['dummy_file.tar.gz', 'dummy_file.tar.gz.sig'])
-        binding.setVariable('UPLOAD_PATH', 'dummy/upload/path')
     }
 
     @Test
     void testPrintArtifactDownloadUrlsForStaging() {
+
+        binding.setVariable('filenamesForUrls', artifactFileNames)
+        binding.setVariable('UPLOAD_PATH', uploadPath)
+
         super.testPipeline("tests/jenkins/jobs/PrintArtifactDownloadUrlsForStaging_Jenkinsfile")
+
+    }
+
+    void configure(helper, bindings){
+    }
+
+    void verifyParams(helper) {
+        assert helper.callStack.findAll { call ->
+            call.methodName == 'printArtifactDownloadUrlsForStaging'
+        }.size() > 0
+
+        helper.callStack.findAll { call ->
+            call.methodName == 'printArtifactDownloadUrlsForStaging'
+        }.each { call ->
+            assertThat(call.args.artifactFileNames.first(), notNullValue())
+            assertThat(call.args.uploadPath.first(), notNullValue())
+            assert call.args.artifactFileNames.size() > 0
+        }
+
+        def callFound = false
+
+        def callList = helper.callStack.findAll { call ->
+            call.methodName == 'printArtifactDownloadUrlsForStaging'
+        }
+
+        for(call in callList){
```

Comment:
I think this doesn't need to be a for loop instead you should be able to write this as

```
assertThat(calllist.Any(call -> {
    call.args.uploadPath.first() == this.uploadPath
       && call.args.artifactFileNames.first().sort() == this.artifactFileNames.sort() 
})).isTrue()
```
*This is assumed I did not test writing this in groovy*